oauth2_http/java/com/google/auth/oauth2/OAuth2Credentials.java

```diff
@@ -307,7 +299,9 @@ private void finishRefreshAsync(ListenableFuture<OAuthValue> finishedTask) {
    * thread of whatever executor the async call used. This doesn't affect correctness and is
    * extremely unlikely.
    */
-  private static <T> T unwrapDirectFuture(Future<T> future) throws IOException {
+  private static <T> T unwrapDirectFuture(ListenableFuture<T> future) throws IOException {
+    future = Futures.transform(future, Functions.identity(), MoreExecutors.directExecutor());
```

Comment:
This shouldn’t be necessary. You are already enforcing ordering in the refreshtask

Reply:
Not familiar with what this line does.

Sounds like I can go ahead and delete it?